app/src/main/java/com/github/sdpcoachme/MainActivity.kt

```diff
@@ -4,53 +4,112 @@ import android.content.Intent
 import android.os.Bundle
 import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
+import androidx.activity.result.ActivityResultLauncher
 import androidx.compose.foundation.layout.Arrangement
 import androidx.compose.foundation.layout.Column
 import androidx.compose.foundation.layout.fillMaxSize
 import androidx.compose.material.Button
 import androidx.compose.material.Text
-import androidx.compose.material.TextField
-import androidx.compose.runtime.*
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.stringResource
+import com.firebase.ui.auth.FirebaseAuthUIActivityResultContract
+import com.github.sdpcoachme.firebase.auth.Authenticator
+import com.github.sdpcoachme.firebase.database.Database
 import com.github.sdpcoachme.ui.theme.CoachMeTheme
 
 class MainActivity : ComponentActivity() {
+
+    private lateinit var database : Database
+    private var signInInfo: String by mutableStateOf("Not signed in")
+    private lateinit var authenticator: Authenticator
+    private val signInLauncher: ActivityResultLauncher<Intent> = registerForActivityResult (
+        FirebaseAuthUIActivityResultContract()
+    ) { res ->
+        authenticator.onSignInResult(
+            res,
+            { email -> launchPostLoginActivity(email ?: "") },
+            { errorMsg -> signInInfo = errorMsg.toString() }
+        )
+    }
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+        authenticator = (application as CoachMeApplication).authenticator
+        this.database =  (application as CoachMeApplication).database
```

Comment:
This is great a very clean use of dependency injection and will merge easily in the project as it was coded the same way in the branch implementing the database.